src/browser/setupWorker/start/createStartHandler.ts

```diff
@@ -13,131 +14,158 @@ export const createStartHandler = (
 
 $ npx msw init <PUBLIC_DIR>
 
 This is necessary to ensure that the Service Worker is in sync with the library to guarantee its stability.
-If this message still persists after updating, please report an issue: https://github.com/open-draft/msw/issues\
+If this message still persists after updating, please report an issue: https://github.com/mswjs/msw/issues\
       `)
-      }
+        }
 
-      context.keepAliveInterval = window.setInterval(
-        () => context.workerChannel.send('KEEPALIVE_REQUEST'),
-        5000,
-      )
+        context.keepAliveInterval = window.setInterval(
+          () => context.workerChannel.send('KEEPALIVE_REQUEST'),
+          5000,
+        )
 
-      // Warn the user when loading the page that lies outside
-      // of the worker's scope.
-      validateWorkerScope(registration, context.startOptions)
+        // Warn the user when loading the page that's outside
+        // of the worker's scope.
+        validateWorkerScope(registration, context.startOptions)
 
-      return registration
-    }
+        return registration
+      }
 
-    const workerRegistration = startWorkerInstance().then(
-      async (registration) => {
-        const pendingInstance = registration.installing || registration.waiting
-
-        // Wait until the worker is activated.
-        // Assume the worker is already activated if there's no pending registration
-        // (i.e. when reloading the page after a successful activation).
-        if (pendingInstance) {
-          await new Promise<void>((resolve) => {
-            pendingInstance.addEventListener('statechange', () => {
-              if (pendingInstance.state === 'activated') {
-                return resolve()
-              }
-            })
-          })
+    registerWorker().then((pendingWorkerRegistration) => {
+      // If the activation was cancelled (e.g. by calling "worker.stop()"),
+      // short-circuit: nothing to return.
+      if (pendingWorkerRegistration == null) {
+        return
+      }
+
+      const pendingWorker =
+        pendingWorkerRegistration.installing ||
+        pendingWorkerRegistration.waiting
+
+      // Wait until the worker is activated.
+      // Assume the worker is already activated if there's no pending registration
+      // (i.e. when reloading the page after a successful activation).
+      if (pendingWorker) {
+        const stateChangeListener = () => {
+          if (pendingWorker.state === 'activated') {
+            workerActivationPromise.resolve(pendingWorkerRegistration)
+
+            pendingWorker.removeEventListener(
+              'statechange',
+              stateChangeListener,
+            )
+          }
         }
 
-        // Print the activation message only after the worker has been activated.
-        await enableMocking(context, options).catch((error) => {
+        pendingWorker.addEventListener('statechange', stateChangeListener)
+      }
```

Comment:
```suggestion
      }
      workerActivationPromise.resolve(pendingWorkerRegistration)
```

I think we need this line here to catch this?

Reply:
if not, we may need some other way to handle this because fresh page loads aren't ever falling into this state change handler, which means that the `enableMocking` calls are never landing.